rmf_traffic/test/unit/test_Profile.cpp

```diff
@@ -0,0 +1,251 @@
+/*
+ * Copyright (C) 2020 Open Source Robotics Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+*/
+
+#include <rmf_traffic/Profile.hpp>
+#include <rmf_traffic/DetectConflict.hpp>
+#include <rmf_traffic/geometry/Circle.hpp>
+#include <rmf_traffic/geometry/Box.hpp>
+
+#include <rmf_utils/catch.hpp>
+
+using namespace std::chrono_literals;
+
+SCENARIO("Testing Construction")
+{
+  using Profile = rmf_traffic::Profile;
+
+  WHEN("Vicinity is not passed into the constructor")
+  {
+    const auto circle_1 = rmf_traffic::geometry::make_final_convex<
+        rmf_traffic::geometry::Circle>(1.0);
+    const auto circle_2 = rmf_traffic::geometry::make_final_convex<
+        rmf_traffic::geometry::Circle>(2.0);
+
+    auto profile = Profile{circle_1};
+
+    auto& footprint = profile.footprint();
+    auto& vicinity = profile.vicinity();
+
+    CHECK((footprint->get_characteristic_length() - 1.0) == Approx(0.0).margin(1e-6));
```

Comment:
Is there a reason to test the difference instead of just testing the value directly? E.g.:

```
CHECK(footprint->get_characteristic_lenght() == Approx(1.0));
```

In other tests we used distance because it's simpler than checking each component, but that's not relevant here since we're testing a scalar value. And I think we can use the default margin instead of specifying one explicitly.

Reply:
You can also consider testing the identity of the footprint.

```
CHECK(footprint == circle_1);
```

should be a valid test here.

Reply:
Thanks for pointing these out. I have modified the checks to test the values directly. Also included checks to test the identity of footprints and vicinities. But not sure if it makes sense to have both though? I could remove the tests for `characteristic_length` if its redundant.